pkg/cmd/adm/restart_test.go

```diff
@@ -1,215 +1,345 @@
-				return fmt.Errorf("some error")
-			}
-			ctx := clicontext.NewCommandContext(term, newClient)
-
-			// when
-			err := restart(ctx, clusterName, "cool-deployment")
-
-			// then
-			require.Error(t, err)
-			fakeClient.MockGet = nil
-			AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 3)
-			assert.Equal(t, 0, numberOfUpdateCalls)
+		//when
+		err := restartDeployments(ctx, fakeClient, HostNamespacedName.Namespace, func(ctx *clicontext.CommandContext, deployment appsv1.Deployment) error {
+			return checkRolloutStatus(ctx, tf, streams, *hostDep)
+		}, func(ctx *clicontext.CommandContext, deployment appsv1.Deployment) error {
+			return restartNonOlmDeployments(ctx, deployment, tf, streams)
 		})
 
-		t.Run("restart fails - deployment not found for "+clusterName, func(t *testing.T) {
-			// given
-			deployment := newDeployment(namespacedName, 3)
-			newClient, fakeClient := NewFakeClients(t, deployment)
-			numberOfUpdateCalls := 0
-			fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 3, &numberOfUpdateCalls)
-			term := NewFakeTerminalWithResponse("Y")
-			ctx := clicontext.NewCommandContext(term, newClient)
-
-			// when
-			err := restart(ctx, clusterName, "wrong-deployment")
-
-			// then
-			require.NoError(t, err)
-			AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 3)
-			assert.Equal(t, 0, numberOfUpdateCalls)
-			assert.Contains(t, term.Output(), "ERROR: The given deployment 'wrong-deployment' wasn't found.")
-			assert.Contains(t, term.Output(), fmt.Sprintf("Existing deployments in toolchain-%s-operator namespace", clusterType))
-			assert.Contains(t, term.Output(), "cool-deployment")
+		//then
+		require.Error(t, err, "no operator based deployment found in namespace toolchain-host-operator , hence no restart happened")
+		assert.Equal(t, 0, csCalls)
+
+	})
+	t.Run("No Non-OLM deployment", func(t *testing.T) {
+		csCalls = 0
+		newClient, fakeClient := NewFakeClients(t, hostDep, pod)
+		ctx := clicontext.NewCommandContext(term, newClient)
+
+		//when
+		err := restartDeployments(ctx, fakeClient, HostNamespacedName.Namespace, func(ctx *clicontext.CommandContext, deployment appsv1.Deployment) error {
+			return checkRolloutStatus(ctx, tf, streams, *hostDep)
+		}, func(ctx *clicontext.CommandContext, deployment appsv1.Deployment) error {
+			return restartNonOlmDeployments(ctx, deployment, tf, streams)
 		})
-	}
+
+		//then
+		require.NoError(t, err)
+		//checking the logic when only operator based deployment is there and no non-operator based
+		require.Contains(t, term.Output(), "No Non-OLM deployment found in namespace toolchain-host-operator, hence no restart happened")
+		assert.Equal(t, 1, csCalls)
+
+	})
+
 }
+func TestRestartDeployment(t *testing.T) {
+	//given
+	SetFileConfig(t, Host(), Member())
+	toolchainCluster := NewToolchainCluster(ToolchainClusterName("host"))
```

Comment:
could you please clarify why there is the `toolcahinCluster`? what is it used for? it shouldn't be needed for the restart tests